src/Core/Grid/Query/LinkBlockQueryBuilder.php

```diff
@@ -0,0 +1,121 @@
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/AFL-3.0 Academic Free License 3.0 (AFL-3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\Module\LinkList\Core\Grid\Query;
+
+use Doctrine\DBAL\Query\QueryBuilder;
+use PrestaShop\PrestaShop\Core\Grid\Query\AbstractDoctrineQueryBuilder;
+use PrestaShop\PrestaShop\Core\Grid\Query\DoctrineQueryBuilderInterface;
+use PrestaShop\PrestaShop\Core\Grid\Search\SearchCriteriaInterface;
+
+/**
+ * Class LinkBlockQueryBuilder
+ * @package PrestaShop\Module\LinkList\Core\Grid\Query
+ */
+final class LinkBlockQueryBuilder extends AbstractDoctrineQueryBuilder
+{
+    /**
+     * @param null|SearchCriteriaInterface $searchCriteria
+     * @return QueryBuilder
+     */
+    public function getSearchQueryBuilder(SearchCriteriaInterface $searchCriteria = null)
+    {
+        $qb = $this->getQueryBuilder($searchCriteria->getFilters());
+        $qb->select('
+            lb.id_link_block,
+            lbl.name AS block_name,
+            lb.id_hook,
+            h.name as hook_name,
+            h.title as hook_title,
+            h.description as hook_description,
+            lb.position
+            ')
+            ->orderBy(
+                $searchCriteria->getOrderBy(),
+                $searchCriteria->getOrderWay()
+            )
+        ;
+
+        if ($searchCriteria->getLimit() > 0) {
+            $qb
+                ->setFirstResult($searchCriteria->getOffset())
+                ->setMaxResults($searchCriteria->getLimit())
+            ;
+        }
+
+        return $qb;
+    }
+
+    /**
+     * @param null|SearchCriteriaInterface $searchCriteria
+     * @return QueryBuilder
+     */
+    public function getCountQueryBuilder(SearchCriteriaInterface $searchCriteria = null)
+    {
+        $qb = $this->getQueryBuilder($searchCriteria->getFilters());
+        $qb->select('COUNT(lb.id_link_block)');
+
+        return $qb;
+    }
+
+    /**
+     * Get generic query builder.
+     *
+     * @param array $filters
+     *
+     * @return QueryBuilder
+     */
+    private function getQueryBuilder(array $filters)
+    {
+        $qb = $this->connection
+            ->createQueryBuilder()
+            ->from($this->dbPrefix . 'link_block', 'lb')
+            ->innerJoin('lb', $this->dbPrefix . 'link_block_lang', 'lbl', 'lb.id_link_block = lbl.id_link_block')
+            ->leftJoin('lb', $this->dbPrefix . 'hook', 'h', 'lb.id_hook = h.id_hook');
+
+        foreach ($filters as $name => $value) {
+            if ('id_lang' === $name) {
```

Comment:
you can remove foreach and edit your if conditions 
```
if (isset($filters['id_lang'])):
if (isset($filters['id_hook'])):
```

or
```php
foreach (['id_lang', 'id_hook'] as $name) {
  if (!isset($filters[$name])) {
    continue;
  }

  $qb
    ->andWhere(sprintf('h.%s = :%s', $name, $name))
    ->setParameter($name, $filters[$name]);
}

Reply:
I inspired myself from the existing query builders from the Core, I think the aim of the loop here is that if someone extends the definition and adds a field or a filter there is no need to update the Query builder hence the loop
@mickaelandrieu do you confirm?

Reply:
But with what you did you're only authorized to have id_lang and id_hook too =)

Reply:
that's quite pertinent! :D